### PR TITLE
eng, update typespec-java to 0.8.10

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
code change https://github.com/Azure/azure-sdk-for-java/pull/36025/commits/fa3f4ab0699e2aa9d413ab75f3f8fd7b0eddef28, reason https://github.com/Azure/azure-sdk-for-java/pull/36025#discussion_r1295426455

emitter change
- include feature of "only generate protocol API, if that operation contains Union"
- support `EnumMember` as type

we skipped 0.8.9, as it has a good feature for Jackson de-serialization, but hit a spotbugs error.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3008065&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=ba3cf43a-b2bf-50f9-f95f-93049ca44ad1

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by srnagar — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
Will wait for OpenAI to remove the `@Generated` annotation on convenient API.

```java
    @Generated
    @ServiceMethod(returns = ReturnType.SINGLE)
    public ChatCompletions getChatCompletions(
```

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 1]
```diff
@@ -423,7 +423,6 @@ public Flux<Completions> getCompletionsStream(String deploymentOrModelName, Comp
      * @return chat completions for the provided chat messages. Completions support a wide variety of tasks and generate
      *     text that continues from or "completes" provided prompt data on successful completion of {@link Mono}.
      */
-    @Generated
```
Comment (weidongxu-microsoft) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:426]: Dev need to maintain this 2 APIs, as

```
Convenience API is not generated, as operation 'getChatCompletions' refers Union 'FunctionCallPreset | FunctionName'
```

Comment (jpalvarezl) [reply sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:426]: I am not entirely sure we want to remove these annotations from these convenience methods. Could we simply update the version of autorest in this PR and then we evaluate how we handle methods exposing types with `union` on our side?

**Review by jpalvarezl — CHANGES_REQUESTED** [review]
We are evaluating how to best approach this. We don't want to use custom where we can avoid it, so if you could revert the removal of `@Generated`, we will see how we can adjust the TSP definition to work around this.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java (hunk 1 above)
Comment (jpalvarezl) [reply sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:426]:
```
Convenience API is not generated, as operation 'getChatCompletions' refers Union 'FunctionCallPreset | FunctionName'
```

Was this a side effect of updating the autorest version?

Comment (jpalvarezl) [reply sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:426]: Srikanta explained this change in behaviour offline to me! Sorry for the confusion. Will merge accordingly.

**Comment by weidongxu-microsoft** [comment]
In future (maybe next month), this `access` PR would get into TCGC https://github.com/Azure/typespec-azure/issues/3322

After that (and the update in emitter), you can write `access` to some of the models that current skipped (due to Union in API) and have them generated. This should help keeping the model in TypeSpec and SDK in sync.

**Pull request merged**